### PR TITLE
Create a substater with an `extraPrefix`

### PR DESCRIPTION
**Pull request by paulmach**
Sometimes you create a RemoteClient with a prefix of "project_name".
You then may want to pass this stater into other objects adding a prefix
identifing the that object is being used. This PR allows for a such a
"substater" to be created using the same underlying connection as the original.

@mlerner

### CODE REVIEW

File: statsd.go [hunk 1]
```diff
@@ -108,6 +110,26 @@ func New(address string, prefix ...string) (Stater, error) {
 	return client, nil
 }
 
+// Substater returns another RemoteClient using the same connection but appending
+// to the stats prefix. A dot (.) will be added between the currenct prefix and extraPrefix
```
Comment (mlerner) [inline statsd.go]: s/currenct/current

File: statsd_test.go [hunk 2]
```diff
@@ -95,7 +99,44 @@ func TestClose(t *testing.T) {
...
 		t.Error("closed connection, should have returned ConnectionClosedErr")
 	}
+}
+
+func TestMultiClose(t *testing.T) {
+	c, err := New("0.0.0.0:1000")
+	if err != nil {
+		t.Fatal(err)
+	}
+
```
Comment (mlerner) [inline statsd_test.go]: This test is to make sure that calling ```Close()``` will not panic?

Comment (paulmach) [inline statsd_test.go]: when you call close multiple times. Code will panic if you try to close a closed channel. so there should be logic to deal with that, at least that is what the test is enforcing.

Comment (mlerner) [inline statsd_test.go]: :+1:

**Pull request merged**